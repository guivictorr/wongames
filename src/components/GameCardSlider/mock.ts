export default [
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: 235.0,
    promotionalPrice: 215.0,
    slug: 'population-zero'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x141',
    price: 235.0,
    promotionalPrice: 215.0,
    slug: 'population-zero'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x142',
    price: 235.0,
    promotionalPrice: 215.0,
    slug: 'population-zero'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x143',
    price: 235.0,
    promotionalPrice: 215.0,
    slug: 'population-zero'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x144',
    price: 235.0,
    promotionalPrice: 215.0,
    slug: 'population-zero'
  },
  {
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://source.unsplash.com/user/willianjusten/300x145',
    price: 235.0,
    promotionalPrice: 215.0,
    slug: 'population-zero'
  }
]
