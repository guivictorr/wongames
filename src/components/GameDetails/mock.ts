import { GameDetailsProps } from '.'

export default {
  platforms: ['windows', 'mac', 'linux'],
  developer: 'Valve',
  publisher: 'Valve',
  genres: ['Action', 'Adventure', 'Strategy'],
  releaseDate: '2020-11-21T23:00:00.000',
  rating: 'BR0'
} as GameDetailsProps
