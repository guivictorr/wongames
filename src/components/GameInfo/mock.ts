export default {
  title: 'Borderlands 3',
  description:
    'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quidem doloremque dolores, quas quos dolore, voluptatem doloribus quae quaerat, voluptas dolorum quisquam doloremque.',
  price: '200.00'
}
