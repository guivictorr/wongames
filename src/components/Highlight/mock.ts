export default {
  title: 'Red dead redemption 2',
  subtitle: 'Come see Johnâ€™s new adventures',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2',
  backgroundImage: '/img/red-dead-img.jpg',
  floatImage: '/img/red-dead-float.png'
}
