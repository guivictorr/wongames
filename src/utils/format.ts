export const formatNumber = (
  value: number,
  options: Intl.NumberFormatOptions
) => {
  return new Intl.NumberFormat('en-US', options).format(value)
}
