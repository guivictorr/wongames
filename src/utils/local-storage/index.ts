const APP_KEY = 'WONGAMES'

export const getStorageItem = (key: string) => {
  if (typeof window === 'undefined') return

  const item = window.localStorage.getItem(`${APP_KEY}_${key}`)

  return JSON.parse(item!)
}

export const setStorageItem = (key: string, value: string | string[]) => {
  if (typeof window === 'undefined') return

  return window.localStorage.setItem(`${APP_KEY}_${key}`, JSON.stringify(value))
}
