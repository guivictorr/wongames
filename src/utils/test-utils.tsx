import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderOptions } from '@testing-library/react'
import {
  CartContextDefaultValues,
  CartContext,
  CartContextData
} from 'hooks/useCart'
import theme from 'styles/theme'

type CustomRenderProps = {
  cartProviderProps?: CartContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  {
    cartProviderProps = CartContextDefaultValues,
    ...renderOptions
  }: CustomRenderProps = {}
) => {
  return render(
    <ThemeProvider theme={theme}>
      <CartContext.Provider value={cartProviderProps}>
        {ui}
      </CartContext.Provider>
    </ThemeProvider>,
    renderOptions
  )
}

export * from '@testing-library/react'
export { customRender as render }
