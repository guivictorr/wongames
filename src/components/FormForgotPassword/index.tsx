import Button from 'components/Button'
import { FormError, FormLoading, FormWrapper } from 'components/Form'
import TextField from 'components/TextField'
import { signIn } from 'next-auth/client'
import { useRouter } from 'next/router'
import { FormEvent, useState } from 'react'
import { Email } from 'styled-icons/material-outlined'
import { FieldErrors, forgotValidate } from 'utils/validators'

const FormForgotPassword = () => {
  const [formError, setFormError] = useState('')
  const [fieldErrors, setFieldErrors] = useState<FieldErrors>({})
  const [values, setValues] = useState({
    identifier: ''
  })
  const [loading, setLoading] = useState(false)
  const { push, query } = useRouter()

  const handleInput = (field: string, value: string) => {
    setValues((prevState) => ({ ...prevState, [field]: value }))
  }

  const handleSubmit = async (event: FormEvent) => {
    event.preventDefault()
    setLoading(true)
    setFormError('')

    const errors = forgotValidate({ email: values.identifier })

    if (Object.keys(errors).length) {
      setFieldErrors(errors)
      setLoading(false)
      return
    }

    setFieldErrors({})

    const result = await signIn('credentials', {
      ...values,
      redirect: false,
      callbackUrl: `${window.location.origin}${query.callbackUrl || ''}`
    })

    if (result?.url) {
      return push(result.url)
    }

    setLoading(false)

    setFormError('username or password is invalid')
  }
  return (
    <FormWrapper>
      {!!formError && <FormError>{formError}</FormError>}
      <form onSubmit={handleSubmit}>
        <TextField
          name="email"
          placeholder="Email"
          type="email"
          error={fieldErrors?.email}
          onInputChange={(value) => handleInput('identifier', value)}
          icon={<Email />}
        />
        <Button type="submit" size="large" fullWidth disabled={loading}>
          {loading ? <FormLoading /> : 'Send email'}
        </Button>
      </form>
    </FormWrapper>
  )
}

export default FormForgotPassword
