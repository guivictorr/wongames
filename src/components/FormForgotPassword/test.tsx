import 'server.mock'
import userEvent from '@testing-library/user-event'
import { render, screen } from 'utils/test-utils'
import FormForgotPassword from '.'

let query = {}
// eslint-disable-next-line @typescript-eslint/no-var-requires
const routerMock = jest.spyOn(require('next/router'), 'useRouter')

routerMock.mockImplementation(() => ({
  query
}))

describe('<FormForgotPassword />', () => {
  it('should render the form', () => {
    const { container } = render(<FormForgotPassword />)
    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /send email/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should validate the email', async () => {
    render(<FormForgotPassword />)
    userEvent.type(screen.getByPlaceholderText(/email/i), 'valid@gmail.com')

    userEvent.click(screen.getByRole('button', { name: /send email/i }))

    expect(
      await screen.findByText(/You just received an email!/i)
    ).toBeInTheDocument()
  })

  it('should show an invalid email', async () => {
    render(<FormForgotPassword />)
    userEvent.type(screen.getByPlaceholderText(/email/i), 'invalid')

    userEvent.click(screen.getByRole('button', { name: /send email/i }))

    expect(
      await screen.findByText(/must be a valid email/i)
    ).toBeInTheDocument()
  })

  it('should show error if email does not exist', async () => {
    render(<FormForgotPassword />)
    userEvent.type(screen.getByPlaceholderText(/email/i), 'fail@gmail.com')

    userEvent.click(screen.getByRole('button', { name: /send email/i }))

    expect(
      await screen.findByText(/this email does not exist/i)
    ).toBeInTheDocument()
  })

  it('should autofill if comes via logged user', async () => {
    query = { email: 'valid@gmail.com' }
    render(<FormForgotPassword />)
    expect(screen.getByPlaceholderText(/email/i)).toHaveValue('valid@gmail.com')

    userEvent.click(screen.getByRole('button', { name: /send email/i }))

    expect(
      await screen.findByText(/You just received an email!/i)
    ).toBeInTheDocument()
  })
})
