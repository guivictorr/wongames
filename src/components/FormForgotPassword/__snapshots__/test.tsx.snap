// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormForgotPassword /> should render the form 1`] = `
.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: linear-gradient( 178.59deg,#ff5f5f -14.51%,#f062c0 102.86%,#f23131 102.86% );
  cursor: pointer;
  border-radius: 0.4rem;
  border: 0;
  padding: 0.8rem;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c7:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c2:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c5 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2.2rem;
  color: #8F8F8F;
}

.c3 > svg {
  width: 100%;
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c6 {
  margin: 3.2rem auto 1.6rem;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<main
  class="c0"
>
  <form>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="email"
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      class="c6 c7"
      type="submit"
    >
      <span>
        Send email
      </span>
    </button>
  </form>
</main>
`
